File: Selenium Web Automation/src/main/java/LoginPage.java

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends Page {

    private final By usernameField = By.id("user-name");
    private final By passwordField = By.id("password");
    private final By loginButton = By.id("login-button");
    private final By loginErrorMsg = By.xpath("//*[@id='login_button_container']/div/form/div[3]/h3");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public void enterUsername(String username) {
        driver.findElement(usernameField).sendKeys(username);
    }

    public void enterPassword(String password) {
        driver.findElement(passwordField).sendKeys(password);
    }

    public void clickLoginButton() {
        driver.findElement(loginButton).click();
    }

    public void open() {
        super.open("");
    }

    public String invalidCredential() {
        return driver.findElement(loginErrorMsg).getText();
    }
}

File: WDIO Cucumber Web Automation/features/pageobjects/login.page.js
import { $ } from '@wdio/globals'
import Page from './page.js';

class LoginPage extends Page {
    get inputUsername () {
        return $('#user-name');
    }

    get inputPassword () {
        return $('#password');
    }

    get buttonSubmit () {
        return $('input[type="submit"]');
    }

    get errorMessage() {
        return $('//*[@id="login_button_container"]/div/form/div[3]/h3');
    }

    async login (username, password) {
        await this.inputUsername.waitForDisplayed({ timeout: 5000 });
        await this.inputUsername.setValue(username);
        await this.inputPassword.setValue(password);
        await this.buttonSubmit.click();
    }

    async validateWrongPasswordDisplayed() {
        await expect(this.errorMessage).toHaveText(
            'Epic sadface: Username and password do not match any user in this service'
        );
    }

    open () {
        return super.open("");
    }
}

export default new LoginPage();
